assets/js/modules/analytics/util/report-validation.js

```diff
@@ -57,3 +57,18 @@ export function isValidDimensions( dimensions ) {
 		return dimension.hasOwnProperty( 'name' ) && typeof dimension.name === 'string';
 	} );
 }
+
+/**
+ * Verifies provided dimensionFilters to make sure they match allowed values found in dimensions.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} dimensionFilters The dimension filters to check.
+ * @return {boolean} TRUE if dimension filters are valid, otherwise FALSE.
+ */
+export function isValidDimensionFilters( dimensionFilters ) {
+	// Ensure every dimensionFilter key corresponds to a valid dimension.
+	return Object.keys( dimensionFilters ).every(
+		( dimension ) => typeof dimensionFilters[ dimension ] === 'string'
+	);
```

Comment:
We shouldn't require dimension values to be a string. They just need to be scalar values (probably we could check whether it's either a string or a number). Something more important to cover in the validation here though is to ensure that a map of `dimensionName => dimensionValue` is passed. The keys here actually need to be strings. For example someone shouldn't be able to pass an array of values.

Reply:
I figured dimension values would be string due to https://developers.google.com/analytics/devguides/reporting/core/v4/rest/v4/reports/batchGet#DimensionFilter ( expressions[] are 'string' )

Reply:
I've added a check for both string and number per your comment, and checking that keys are string.